src/EventRegistry.h

```diff
@@ -99,6 +100,16 @@ class EventRegistry final {
      */
     EventGroupPtr LookupGroup(EventGroupKind kind, std::string_view name);
 
+    /**
+     * Registers an event handler that calls a C++ std::function instead of a script-land event handler.
+     *
+     * @param name The name of the event handler to lookup/register.
+     * @param priority The priority of the function body in the handler.
+     * @param f The std::function or lambda to call when the event is handled.
+     * @return true if the registration was successful, false otherwise.
+     */
+    bool Register(std::string_view name, int priority, std::function<void(const zeek::Args&)> f);
```

Comment:
IMO, I don't think we should have that API on the registry. Registry is to get handlers, not necessarily containing logic to modify them. The `Register()` behavior here also disagrees with the existing Register() method which actually installs a handler if it doesn't exist.

https://github.com/zeek/zeek/pull/3643/files#diff-3b5342aaebade88c649afd23944bf7ea51f98eebd0c7f859188cb064ea6ba232R50

To me, if anything, this seems more like a helper/convenience function that could live in utils.h.

```
bool add_std_func_to_handler(std::string_view name, int priority, std::function<void(const zeek::Args&)> f) {
   if ( auto handler = Lookup(name) ) {
        handler->GetFunc()->AddBody(std::move(f), priority);
        return true;
   }
   return false;
}
```

And I could actually see not even offering that unless it ends up being a very often used pattern.

I know the first implementation you had started with this API and maybe you're hanging on to it? :-)

Reply:
I was hanging onto it as a convenience method, but I agree that it's not necessary. I can drop it.